core/plans/microvm_delete.go

```diff
@@ -122,8 +122,11 @@ func (p *microvmDeletePlan) addStep(ctx context.Context, step planner.Procedure)
 func (p *microvmDeletePlan) addNetworkSteps(ctx context.Context, vm *models.MicroVM, networkSvc ports.NetworkService) error {
 	for i := range vm.Spec.NetworkInterfaces {
 		iface := vm.Spec.NetworkInterfaces[i]
+		ifaceStats := vm.Status.NetworkInterfaces[iface.GuestDeviceName]
```

Comment:
It shouldn't happen but is it worth checking that the status was found with `ok`?

Reply:
Actually, reading further down i see the test in the `ShouldDo`, so ignore the above